src/models/game_session.ts

```diff
@@ -338,4 +364,13 @@ export default class GameSession {
     resetLastPlayedSongsQueue() {
         this.lastPlayedSongsQueue = [];
     }
+
+    addParticipant(user: Eris.User) {
+        this.participants.add(user.id);
+        if (this.eliminationMode === GameType.ELIMINATION) {
+            const eliminationScoreboard = this.scoreboard as EliminationScoreboard;
+            eliminationScoreboard.addPlayer(user.id, getUserIdentifier(user), user.avatarURL);
+            this.scoreboard = eliminationScoreboard;
```

Comment:
is this needed?

Reply:
Yup, it's the only place where we can pass through the user's identifier and avatarURL to the elimination scoreboard (we only store users as their IDs in gameSession)

Reply:
I meant re-assigning eliminationScoreboard to this.scoreboard here

Reply:
I had some confusion on how type assertion works, it isn't necessary since it isn't creating a new object